### PR TITLE
UKRIW-150 login spike

### PR DESCRIPTION
**Pull request by KatarzynaPaszko**
POC for authorization spike ticket: https://ukri-spyrosoft.atlassian.net/browse/UKRIW-150

### CODE REVIEW

File: apps/eodh-fe/.env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+VITE_BASE_URL=http://localhost:4200
```
Comment (mlu-ssoft) [inline apps/eodh-fe/.env:1]: remove, this file shouldn't be in the repo

File: apps/eodh-fe/index.html [hunk 2]
```diff
@@ -11,7 +11,7 @@
       content="
         style-src   'self' 'nonce-94bf401884d3eade93c7ab13c45437c5' 'strict-dynamic';
         font-src    'self';
-        script-src  'self';
+        script-src  'self' https://test.eodatahub.org.uk;
```
Comment (mlu-ssoft) [inline apps/eodh-fe/index.html:14]: 1. why we need this?
2. it shouldn't be hardcoded, we can have different environments

File: apps/storybook-host/.env [hunk 3]
```diff
@@ -0,0 +1,3 @@
+VITE_BASE_URL=http://localhost:4200
```
Comment (mlu-ssoft) [inline apps/storybook-host/.env:1]: remove, this file shouldn't be in the repo

File: jest.config.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { getJestProjects } from '@nx/jest';
```
Comment (mlu-ssoft) [inline jest.config.ts:1]: we use vitest for tests, change it

File: jest.preset.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+const nxPreset = require('@nx/jest/preset').default;
```
Comment (mlu-ssoft) [inline jest.preset.js:1]: same here, use vitest

File: libs/authorization/data-access/package.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (mlu-ssoft) [inline libs/authorization/data-access/package.json:1]: don't create buildable libraries, we won't publish it to npm

File: libs/authorization/data-access/src/lib/keycloak.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import Keycloak from 'keycloak-js';
+
+const keycloak = new Keycloak({
+  url: 'https://test.eodatahub.org.uk/keycloak',
```
Comment (mlu-ssoft) [inline libs/authorization/data-access/src/lib/keycloak.ts:4]: those settings should be in .env file

File: libs/authorization/ui/package.json [hunk 8]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (mlu-ssoft) [inline libs/authorization/ui/package.json:1]: remove, we don't need buildable libraries

File: libs/authorization/ui/src/lib/authorization.tsx [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+import React from 'react';
+import { useEffect, useState } from 'react';
+
+import { LoginButton } from './login-button';
+import { LogoutButton } from './logout-button';
+
+export const Login = ({ className }: { className: string }) => {
+  const { keycloak, initialized } = useKeycloak();
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+  const currentUrl = window.location.href;
```
Comment (mlu-ssoft) [inline libs/authorization/ui/src/lib/authorization.tsx:12]: shouldn't you save url in localStorage and then use it?

File: libs/authorization/ui/src/lib/authorization.tsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+  useEffect(() => {
+    if (initialized) {
+      setIsAuthenticated(keycloak.authenticated ?? false);
+    }
+  }, [keycloak, initialized]);
+
+  return (
+    <div className={clsx('flex items-center', className)}>
```
Comment (mlu-ssoft) [inline libs/authorization/ui/src/lib/authorization.tsx:21]: we don't need ui library for authorisation, move it to default layout

File: package.json [hunk 11]
```diff
@@ -51,6 +53,7 @@
     "@nx/cypress": "19.2.3",
     "@nx/eslint": "19.2.3",
     "@nx/eslint-plugin": "19.2.3",
+    "@nx/jest": "16.10.0",
```
Comment (mlu-ssoft) [inline package.json:56]: remove jest

File: libs/authorization/data-access/.eslintrc.json [hunk 12]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (mlu-ssoft) [inline libs/authorization/data-access/.eslintrc.json:1]: move all code from `authorisation/data-access` into `shared/utils/auth`. If we leave this in authorisation then we won't be able to use it in other places (eg. in map `domain`)

File: libs/authorization/data-access/project.json [hunk 13]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "data-access-authorization",
+  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
+  "sourceRoot": "libs/authorization/data-access/src",
+  "projectType": "library",
+  "tags": ["type:data-access"],
```
Comment (mlu-ssoft) [inline libs/authorization/data-access/project.json:6]: scope missing

File: nx.json [hunk 14]
```diff
@@ -29,6 +29,14 @@
     "build-storybook": {
       "cache": true,
       "inputs": ["default", "^production", "{projectRoot}/.storybook/**/*", "{projectRoot}/tsconfig.storybook.json"]
+    },
+    "@nx/js:tsc": {
```
Comment (mlu-ssoft) [inline nx.json:33]: why this was added? do we use tsc executor somewhere?

**Review by mlu-ssoft — CHANGES_REQUESTED** [review]

File: apps/eodh-fe/index.html (hunk 2 above)
Comment (KatarzynaPaszko) [reply apps/eodh-fe/index.html:14]: It was for POC testing purposes. I will remove it

File: libs/authorization/ui/src/lib/authorization.tsx (hunk 9 above)
Comment (KatarzynaPaszko) [reply libs/authorization/ui/src/lib/authorization.tsx]: as discussed in priv, not needed

File: nx.json (hunk 14 above)
Comment (KatarzynaPaszko) [reply nx.json]: removed

**Review by mlu-ssoft — APPROVED** [review]

**Pull request merged**